cmake/FindArpack.cmake

```diff
@@ -0,0 +1,20 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+if(NOT ARPACK_ROOT)
+  # Need to set to empty to avoid warnings with --warn-uninitialized
+  set(ARPACK_ROOT "")
+  set(ARPACK_ROOT $ENV{ARPACK_ROOT})
+endif()
+
+find_library(ARPACK_LIBRARIES
```

Comment:
Could you split the changes that find the Arpack library and make it available in CMake into a separate commit please? It'll also need an entry in `docs/Installation/Installation.md`.

Question: Are you suggesting to make Arpack an optional dependency, or will it be required? It's probably easy enough to install to make it a required dependency, but we should make sure of that.

Reply:
Related: the commits should be verby (bottom of https://spectre-code.org/contributing_to_spectre.html) eg "Add CMake find files for arpack", "Add Arpack to HPC caltech env"